### PR TITLE
New package: reuse-0.13.0

### PR DESCRIPTION
**Pull request by k4leg**
See <https://reuse.software> for more info.

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [ ] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [x] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by Chocimier** [comment]
- no new py2 packages are accepted
- depending on minimal version of module is going to stale, please either provide tool to keep it up to date on update, or drop
- testing with tox undex xbps-src is useless, try good deprecated things
-  `make_check=` has to be justified
- no need do depend on interpreter when depending on module

**Comment by Chocimier** [comment]
- please use pypi as homepage for packages not fetched from pypi, this helps find upstream in case it moves

**Comment by paper42** [comment]
> - please use pypi as homepage for packages not fetched from pypi, this helps find upstream in case it moves

PyPi tarballs also remove tests, so I think it would be nicer to use the non-pypi tarballs when there are tests to run.

**Comment by Chocimier** [comment]
> nicer to use the non-pypi tarballs

Makes sense if this is case for these packages, but don't that depend on setup.py?

**Comment by paper42** [comment]
> > nicer to use the non-pypi tarballs
> 
> Makes sense if this is case for these packages, but don't that depend on setup.py?

setup.py test is the old way of running tests, for modern projects, we can use pytest from python3-pytest.

**Comment by Chocimier** [comment]
I mean boolean.py pypi tarball includes tests.

**Comment by paper42** [comment]
> I mean boolean.py pypi tarball includes tests.

I get your point now. Some projects have their tests in pypi tarballs, some don't, so we could check for each package.

**Comment by k4leg** [comment]
>     * `make_check=` has to be justified

I don't understand why but the tests fail in CI (see [this](https://github.com/void-linux/void-packages/actions/runs/1179827994)). With a local build everything is fine. Perhaps this will change after switching from tox to other.

>     * testing with tox undex xbps-src is useless, try good deprecated things

Btw I saw in some packages tests are missing because some dependencies are missing. Is it okay to do the same or is it better to pack them? This is why I used tox.

**Comment by Chocimier** [comment]
Best is to selectively disable tests that fail for known reason, and report other upstream.

File: srcpkgs/python3-binaryornot/template [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'python3-binaryornot'
+pkgname=python3-binaryornot
+_pkgname=binaryornot
```
Comment (paper42) [inline srcpkgs/python3-binaryornot/template:3]: remove this variable and replace where it's used with `binaryornot` (similar change should also be done in other packages)

File: srcpkgs/python3-binaryornot/template [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+version=0.4.4
+revision=1
+wrksrc="${_pkgname}-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-chardet>=3.0.2"
```
Comment (paper42) [inline srcpkgs/python3-binaryornot/template:9]: please remove the version constraints (similar change should also be done in other packages)

File: srcpkgs/python3-debian/template [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+depends="python3-chardet python3-six"
+checkdepends="dpkg tar $depends"
+short_desc="Python modules work with Debian-related data formats"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="GPL-2.0-or-later"
+homepage="https://pypi.org/project/python-debian"
+# Don't use PyPI because it has no tests.
+distfiles="${homepage}/-/archive/${version}/${_pkgname}-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/python3-debian/template:16]: we prefer not using `$homepage` in `$distfiles`

File: srcpkgs/python3-debian/update [hunk 4]
```diff
@@ -0,0 +1,2 @@
+site='https://salsa.debian.org/python-debian-team/python-debian/tags?format=atom'
+pattern='<title>\K[\d.]+(?=</title>)'
```
Comment (paper42) [inline srcpkgs/python3-debian/update:2]: we can make this prettier:
```
pkgname=python-debian
site=https://salsa.debian.org/python-debian-team/python-debian/-/tags
```

Comment (k4leg) [reply srcpkgs/python3-debian/update:2]: Thanks!

File: srcpkgs/reuse/template [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+depends="python3-binaryornot python3-boolean.py python3-debian python3-Jinja2
+ python3-license-expression python3-requests"
+checkdepends="python3-pytest $depends"
+short_desc="Helper tool for compliance with REUSE Specification"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="GPL-3.0-or-later, CC-BY-SA-4.0, CC0-1.0, Apache-2.0"
+homepage="https://pypi.org/project/reuse"
+distfiles="${PYPI_SITE}/r/reuse/reuse-${version}.tar.gz"
+checksum=89c87bd49308c8059cc36f8116872d73dc6e5788a29af8869be6532202bff801
+make_check=ci-skip  # The local build passes the tests but not the CI build.
```
Comment (Chocimier) [inline srcpkgs/reuse/template:16]: This is a `a=b; // assign b to a` style comment. Could you remove `make_check=ci-skip` for a while to show what's wrong ?

Comment (Chocimier) [reply srcpkgs/reuse/template]: So this is about three cases that assume no root privs. Disable them like in diffoscope template.

Comment (k4leg) [reply srcpkgs/reuse/template]: Thanks!

File: srcpkgs/python3-boolean.py/template [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+version=3.8
+revision=1
+wrksrc="boolean.py-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools"
+short_desc="Small library implementing a boolean algebra"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="BSD-2-Clause"
+homepage="https://github.com/bastikr/boolean.py"
+distfiles="${PYPI_SITE}/b/boolean.py/boolean.py-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/python3-boolean.py/template:12]:
```
changelog="https://raw.githubusercontent.com/bastikr/boolean.py/master/CHANGELOG.rst"
```

File: srcpkgs/python3-license-expression/template [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+wrksrc="license-expression-${version}"
+build_style=python3-module
+hostmakedepends="python3-setuptools python3-setuptools_scm"
+depends="python3-boolean.py"
+checkdepends="python3-pytest $depends"
+short_desc="Library to parse, normalize and compare license expressions"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="Apache-2.0"
+homepage="https://github.com/nexB/license-expression"
+distfiles="${PYPI_SITE}/l/license-expression/license-expression-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/python3-license-expression/template:14]:
```
changelog="https://raw.githubusercontent.com/nexB/license-expression/master/CHANGELOG.rst"
```

File: srcpkgs/reuse/template [hunk 8]
```diff
@@ -0,0 +1,16 @@
+# Template file for 'reuse'
+pkgname=reuse
+version=0.13.0
+revision=1
+build_style=python3-module
+make_check_args="-k not((test_lint_read_errors)or(test_generate_project_report_read_error))"
```
Comment (paper42) [inline srcpkgs/reuse/template:6]: `--deselect=` should also work and it will be more readable

File: srcpkgs/reuse/template [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+hostmakedepends="python3-setuptools python3-setuptools_scm gettext"
+depends="python3-binaryornot python3-boolean.py python3-debian python3-Jinja2
+ python3-license-expression python3-requests"
+checkdepends="python3-pytest $depends"
+short_desc="Helper tool for compliance with REUSE Specification"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="GPL-3.0-or-later, CC-BY-SA-4.0, CC0-1.0, Apache-2.0"
+homepage="https://reuse.software"
+distfiles="${PYPI_SITE}/r/reuse/reuse-${version}.tar.gz"
```
Comment (paper42) [inline srcpkgs/reuse/template:15]:
```
changelog="https://raw.githubusercontent.com/fsfe/reuse-tool/master/CHANGELOG.md"
```

File: srcpkgs/reuse/template [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+revision=1
+build_style=python3-module
+make_check_args="-k not((test_lint_read_errors)or(test_generate_project_report_read_error))"
+hostmakedepends="python3-setuptools python3-setuptools_scm gettext"
+depends="python3-binaryornot python3-boolean.py python3-debian python3-Jinja2
+ python3-license-expression python3-requests"
+checkdepends="python3-pytest $depends"
+short_desc="Helper tool for compliance with REUSE Specification"
+maintainer="k4leg <python.bogdan@gmail.com>"
+license="GPL-3.0-or-later, CC-BY-SA-4.0, CC0-1.0, Apache-2.0"
```
Comment (paper42) [inline srcpkgs/reuse/template:13]: All original source code is licensed under GPL-3.0-or-later.
    All documentation is licensed under CC-BY-SA-4.0.
    Some configuration and data files are licensed under CC0-1.0.
    Some code borrowed from spdx/tool-python is licensed under Apache-2.0.

I think we don't need to list CC* licenses here.

File: srcpkgs/reuse/template (hunk 8 above)
Comment (k4leg) [reply srcpkgs/reuse/template:6]: Thanks!

File: srcpkgs/reuse/template (hunk 10 above)
Comment (k4leg) [reply srcpkgs/reuse/template:13]: Yes you are right. But the package still contains files with license CC0-1.0 (for example `/usr/lib/pythonX.Y/site-packages/reuse/resources/exceptions.json`), so I will only remove license CC-BY-SA-4.0.
Thanks!

**Pull request merged**